support/racket/support.rkt

```diff
@@ -409,14 +409,13 @@
 (define (blodwen-clock-second time) (time-second time))
 (define (blodwen-clock-nanosecond time) (time-nanosecond time))
 
-(define (blodwen-args)
-  (define (blodwen-build-args args)
-    (if (null? args)
-        (vector 0) ; Prelude.List
-        (vector 1 (car args) (blodwen-build-args (cdr args)))))
-    (blodwen-build-args
-      (cons (path->string (find-system-path 'run-file))
-            (vector->list (current-command-line-arguments)))))
```

Comment:
By using `current-command-line-arguments` you accidentally threw away
`path->string (find-system-path 'run-file)` as the first argument.

I expect that somewhere Idris assumes it gets the name of the executable
first & then the args and that breaks that assumption?

You should probably have something like:

```racket
(define (blodwen-arg-count)
  (+ 1 (vector-length (current-command-line-arguments))))

(define (blodwen-arg n)
  (cond 
     [(= 0 n) (path->string (find-system-path 'run-file))]
     [(< (- n 1) (vector-length (current-command-line-arguments)))
         (vector-ref (current-command-line-arguments) (- n 1))]
     [else ""])
```

Reply:
Ok!